backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/controller/UsersController.java

```diff
@@ -25,26 +34,71 @@
 @Tag(name = "Users", description = "CRUD endpoint for users.")
 public class UsersController {
 
+	private static final Logger log = LoggerFactory.getLogger(UsersController.class);
+
 	private final SubscriptionModelAssembler subscriptionModelAssembler;
 
 	private final SubscriptionService subscriptionService;
 
-	public UsersController(SubscriptionModelAssembler subscriptionModelAssembler, SubscriptionService subscriptionService) {
+	private final SubscriptionModelMapper mapper;
+
+	public UsersController(SubscriptionModelAssembler subscriptionModelAssembler, SubscriptionService subscriptionService, SubscriptionModelMapper mapper) {
 		Assert.notNull(subscriptionModelAssembler, "subscriptionModelAssembler is required; it must not be null");
 		Assert.notNull(subscriptionService, "subscriptionService is required; it must not be null");
 		this.subscriptionModelAssembler = subscriptionModelAssembler;
 		this.subscriptionService = subscriptionService;
+		this.mapper = mapper;
 	}
 
-	@GetMapping({ "/{id}/subscriptions" })
+	@GetMapping({ "/{userId}/subscriptions" })
 	@Operation(summary = "Get all subscriptions for a user.", operationId = "get-user-subscriptions")
 	public CollectionModel<SubscriptionModel> getSubscriptionsByUserId(
-			@NotBlank(message = "id must not be null or blank")
-			@Parameter(description = "The id of the user.", example = "000000000", required = true)
-			@PathVariable String id) {
-		final var subscriptions = subscriptionModelAssembler.toCollectionModel(subscriptionService.getSubscriptionsByUserId(id));
-		final var selfLink = linkTo(methodOn(getClass()).getSubscriptionsByUserId(id)).withSelfRel();
+			@NotBlank(message = "user id must not be null or blank")
+			@Parameter(description = "The user id of the user.", example = "000000000", required = true)
+			@PathVariable String userId) {
+		final var subscriptions = subscriptionModelAssembler.toCollectionModel(subscriptionService.getSubscriptionsByUserId(userId));
+		final var selfLink = linkTo(methodOn(getClass()).getSubscriptionsByUserId(userId)).withSelfRel();
 		return subscriptionModelAssembler.wrapCollection(subscriptions, SubscriptionModel.class).add(selfLink);
+	}	
+
+
+	@PostMapping({ "/{userId}/subscriptions/" })
+	@Operation(summary = "Create a subscription for a user.", operationId = "user-subscriptions-create")
+	@ApiResponse(responseCode = "202", description = "The request has been accepted for processing.")
+	public void create(
+			@RequestBody(required = true)
+			SubscriptionModel createSubscriptionRequest,
+
+			@RequestParam(defaultValue = "true", required = false)
+			@Parameter(description = "If the request should be handled asynchronously.", schema = @Schema(allowableValues = { "false", "true" }, defaultValue = "true"))
+			String async) {
```

Comment:
You can remove this since it isn't used in this method.

We sometimes include this async string/boolean to allow the consumer to bypass the queue and process the request immediately, but in this case since we're not using a queue it is unnecessary. We can add it back in a future PR if it's required.

Reply:
removed